### PR TITLE
Added support for GROUP BY in combination with subselect

### PR DESCRIPTION
**Pull request by christianbader**

### CODE REVIEW

File: sql/src/main/java/io/crate/planner/consumer/GroupingSubselectConsumer.java [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+
+        public Visitor(ProjectionBuilder projectionBuilder) {
+            this.projectionBuilder = projectionBuilder;
+        }
+
+        @Override
+        public Plan visitQueriedSelectRelation(QueriedSelectRelation relation, ConsumerContext context) {
+            QuerySpec querySpec = relation.querySpec();
+
+            // Check if group by is present - if not skip and continue with next consumer in chain.
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/planner/consumer/GroupingSubselectConsumer.java]: I'd remove the comment, this pattern should be well-known - or we should rather extend the docs in the Consumer interface or in the ConsumingPlanner.

Another option would be to add class javadocs to this class which describes which kind of statement it handles and what kind of plan it generates

File: sql/src/main/java/io/crate/planner/consumer/GroupingSubselectConsumer.java [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+                where
+            );
+            plan.addProjection(filterProjection, null, null, null);
+        }
+    }
+
+    /**
+     * Returns true for non distributed and false for distributed execution.
+     */
+    private static boolean isExecutedOnHandler(ConsumerContext context, Plan plan) {
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/planner/consumer/GroupingSubselectConsumer.java]: is this abstraction method worth it? Has the same number of arguments and the args are even "bigger" in that pull in classes that are harder to create

File: sql/src/test/java/io/crate/planner/consumer/GroupByPlannerTest.java [hunk 3]
```diff
@@ -618,13 +618,20 @@ public void testGroupByHaving() throws Exception {
 
     @Test
     public void testNestedGroupByAggregation() throws Exception {
```
Comment (mfussenegger) [inline sql/src/test/java/io/crate/planner/consumer/GroupByPlannerTest.java]: Please clarify in the test method what aspect you want to test

File: sql/src/main/java/io/crate/planner/consumer/GroupingSubselectConsumer.java (hunk 2 above)
Comment (christianbader) [inline sql/src/main/java/io/crate/planner/consumer/GroupingSubselectConsumer.java]: you are right - removed it.

File: sql/src/test/java/io/crate/planner/consumer/GroupByPlannerTest.java (hunk 3 above)
Comment (christianbader) [inline sql/src/test/java/io/crate/planner/consumer/GroupByPlannerTest.java]: changed also the assertions

**Pull request merged**